onadata/libs/serializers/fields/project_related_field.py

```diff
@@ -1,15 +1,32 @@
-from onadata.apps.logger.models import Project
+from django.contrib.contenttypes.models import ContentType
+from django.core.cache import cache
+
 from rest_framework import serializers
 from rest_framework.fields import SkipField
 
+from onadata.apps.logger.models import Project
+
 
 class ProjectRelatedField(serializers.RelatedField):
     """A custom field to represent the content_object generic relationship"""
 
     def get_attribute(self, instance):
-        # xform is not an attribute of the MetaData object
-        if instance and isinstance(instance.content_object, Project):
-            return instance.content_object
+        content_type_id = cache.get("project_content_type_id")
```

Comment:
this exactly duplicates 14-29 above and below on lines 15-30, except with a different cache key, can we DRY this up by moving it to a helper function that takes the key as argument?